### PR TITLE
#24 Adjust more config option for Flink component

### LINKED ISSUE #24: Adjust more config option for Flink component
* state backend
* HA
* checkpoint metadata

### PR DESCRIPTION
**Pull request by yanghua**
close #24

### CODE REVIEW

File: hadoop-master1/files/etc/flink/conf/flink-conf.yaml [hunk 1]
```diff
@@ -105,16 +105,18 @@ parallelism.default: 1
...
-# state.backend.type: hashmap
+state.backend.type: hashmap
+
+state.checkpoints.num-retained: 3
 
 # Directory for checkpoints filesystem, when using any of the default bundled
 # state backends.
 #
-# state.checkpoints.dir: hdfs://namenode-host:port/flink-checkpoints
+state.checkpoints.dir: hdfs://hadoop-master1.orb.local:8020/flink-checkpoints
```
Comment (pan3793) [inline hadoop-master1/files/etc/flink/conf/flink-conf.yaml:115]: will flink auto create the folder if not exist?

File: hadoop-master1/files/etc/flink/conf/flink-conf.yaml [hunk 2]
```diff
@@ -60,7 +60,7 @@ parallelism.default: 1
 
 # The high-availability mode. Possible options are 'NONE' or 'zookeeper'.
 #
-# high-availability.type: zookeeper
+high-availability.type: zookeeper
```
Comment (pan3793) [inline hadoop-master1/files/etc/flink/conf/flink-conf.yaml:63]: how does ha work? run two JM in active-standy mode?

File: hadoop-master1/files/etc/flink/conf/flink-conf.yaml (hunk 1 above)
Comment (yanghua) [reply hadoop-master1/files/etc/flink/conf/flink-conf.yaml:115]: > will flink auto create the folder if not exist?

Yes, I have verified.

File: hadoop-master1/files/etc/flink/conf/flink-conf.yaml (hunk 2 above)
Comment (yanghua) [reply hadoop-master1/files/etc/flink/conf/flink-conf.yaml:63]: > how does ha work? run two JM in active-standy mode?

For standalone mode, zk is used for leader election, job graph, checkpoint metadata persistence for failover and recovery.
For flink on yarn, zk is not used for leader election (there is only one JM in yarn's appmaster) and jm discovery but still for storing those metadata.

Comment (pan3793) [reply hadoop-master1/files/etc/flink/conf/flink-conf.yaml:63]: thanks for explanation

**Review by pan3793 — APPROVED** [review]

**Pull request merged**